components/archiving/caseController.js

```diff
@@ -0,0 +1,439 @@
+      res.status(201).end();
+    } else {
+      for (let existingCase of existingCases) {
+        // check for conflicting changes
+        let errors = hasConflictingChanges(existingCase, req.body);
+        if (errors.length) {
+          await archiveDao.setEntityArchiveDoNotProcess(
+            existingCase.entityIdentifier
+          );
+          caseArchiveStopProcessing.inc({
+            caseIdentifier: existingCase.entityIdentifier,
+          });
+          for (const e of errors) {
+            logger.error(e);
+          }
+          throw new ConflictingDataError(errors.join('\n'));
+        }
+        if (!hasArchivingStarted(existingCase)) {
+          // can modify a case that hasn't been archived if there are no errors
+          await upsert(req.body, false);
+          let updatedCase = await archiveDao.getByArchiveEntityIdentifier(
+            req.body.caseIdentifier,
+            false,
+            caseEntityType
+          );
+          res.status(200).send(replaceGenericKeyName(updatedCase));
+          return true;
+        } else {
+          // if case has started archiving, can only modify case metadata
+          await archiveDao.updateMetadata(
+            req.params.caseIdentifier,
+            req.body.metadata
+          );
+          let updatedCase = await archiveDao.getByArchiveEntityIdentifier(
+            req.body.caseIdentifier,
+            false,
+            caseEntityType
+          );
+          res.status(200).send(replaceGenericKeyName(updatedCase));
+          return true;
+        }
+        /* NOTE: if we keep encountering HALP actions that don't require any further fixing other than
+           hitting the case/<caseIdentifier>/resume-archiving endpoint because the underlying discrepancy
+           gets resolved upstream, we might want to adjust these last two conditions to also clear the
+           do-not-process flag */
+      }
+    }
+  } catch (e) {
+    handleErrors(
+      e,
+      `Error adding case ${req.body.caseIdentifier}`,
+      logger,
+      next
+    );
+  }
+};
+
+const upsert = (caseInfo, createNewArchive) => {
+  return archiveDao.addArchiveEntity(
+    replaceCaseKeyName(caseInfo),
+    createNewArchive,
+    caseEntityType
+  );
+};
+
+const filesCopiedToOffsiteStagingDir = async (req, res, next) => {
+  try {
+    let errors = [];
+    if (!req.body.copyOutFile) {
+      errors.push('Must provide a copyOutFile in request body');
+    }
+    if (!req.body.batchId) {
+      errors.push('Must provide a batchId');
+    }
+    if (errors.length) {
+      throw new ValidationError(errors.join('\n'));
+    }
+    const updatedCase = await archiveDao.updateFilesCopiedToOffsiteStagingDir(
+      req.params.caseIdentifier,
+      caseEntityType,
+      req.body.batchId,
+      JSON.stringify(req.body.copyOutFile)
+    );
+    if (updatedCase && updatedCase.length) {
+      res.status(200).json(replaceGenericKeyName(updatedCase));
+    } else {
+      res.status(404).end();
```

Comment:
`updatedCase` will also be empty if the case exists, but `files_copied_to_offsite_archive_staging_dir` is non-null. I'm guessing that's an error, but maybe it's ok? Either way, it shouldn't be 404. Below functions have similar cases too.